### ISSUE TITLE
META: Pair programming; research & disseminate, and then break into tazks and discussions as needed

### ISSUE THREAD

**Issue by serundeputy** [thinktandem/tandem#5]
Pair Programming
================

- Research pair programming

  - take notes

  - summarize & disseminate

    - blog??

    - Lacroix & Learn?? / Friday Meeting

**Comment by serundeputy** [comment]
@pirog see if you think this is good enough for the big time.

Pair Programming
================

Why, what, and how to Pair Program at Tandem.

Let's start by finding interested parties and pair for 2 hours per week.


Why
---

Why are we pair programming?

  * Happy Engineers
    * pairing will help you learn and explore
    * Engineers like to talk about code pairing gives a focused opportunity to do just that
  * Better code
    * optimized/robust/reuseable code
    * testable/tested code
  * Learning on the part of the Engineers
  * Repeatability/robustness of the team
    * often whoever writes a piece of the codebase becomes the owner and only person who understands this piece of code
     * pairing can allow any engineer to work on any part of the code (over time)

What
----

What will we pair program?

There are two use cases at Tandem for pairing:

  * Project code
    * Pick a specific card or feature set for the session
  * Professional Development
    * Pick a specific tech to expand or new tech to learn
    * In this scenario take good notes about paths tried, what worked, what didn't, and a conclusion about the usefulness of this tech at Tandem
  * Don't be dogmatic about tooling
    * let the driver use whatever code editor they like etc.

How
---

How to accomplish pairing for maximum individual and Tandem benefit?

  * schedule it like a meeting
    * spin up a video chat
    * split the time of driver/observer relatively equally
      * that said the fundamental philosophy here is that both parties are engaged
        * if you find your engagement slipping; interject and say 'hey can I drive for 15min or so?'
        * if you need a break don't start checking email; ask for a break; take 5 and go for a little walk
    * be consistent pair for 2hrs/week every week
  * pick a specific card or feature set to work on
  * during pairing
    * do not check or respond to emails or IMs
  * focus on task at hand whilst one types/drives the other should help, debug, question, and think about testablilty

Resources
---------

* [Pair_programming](https://en.wikipedia.org/wiki/Pair_programming)
* [How to Pair Program](https://medium.com/@jdxcode/how-to-pair-program-d6741077e513)

**Comment by pirog** [comment]
@serundeputy im going to classify this as a "note". We decided to change the process on Friday after everyone had a chance to deliver feedback. `tl;dr` we made it simpler. I'd re-review the README and step through the new docz
https://github.com/thinktandem/tandem

Might also be worth checking out the board and seeing how ive spun up a few discussions and tasks to get a more concrete idea of how that all works. Once you do i'd recommend breaking this note up into relevant discussions/tasks
